src/plugins/dashboard/public/locator.ts

```diff
@@ -80,32 +86,47 @@ export interface DashboardAppLocatorParams extends SerializableRecord {
   /**
    * List of dashboard panels
    */
-  panels?: SavedDashboardPanel[] & SerializableRecord;
+  panels?: SavedDashboardPanel[];
 
   /**
    * Saved query ID
    */
   savedQuery?: string;
 }
 
-export type DashboardAppLocator = LocatorPublic<DashboardAppLocatorParams>;
+export type SerializableDashboardAppLocatorParams = DashboardAppLocatorParams & SerializableRecord;
```

Comment:
you shouldn't `& SerializableRecord`, this makes SerializableDashboardAppLocatorParams too general (you can asign any property to it, you loose all the type safety)